cmd/crossplane/core/core.go

```diff
@@ -59,7 +69,35 @@ func (c *Command) Run(log logging.Logger) error {
 		return errors.Wrap(err, "Cannot get config")
 	}
 
-	mgr, err := ctrl.NewManager(cfg, ctrl.Options{SyncPeriod: &c.Sync})
+	// approach 1:
+	// abstract away `ctrl.NewManager` call in crossplane-runtime
+	// ----------------------------------------------------------
+	// mgr, err := controller.NewManager(cfg, controller.Options{
+	// 	Options: &ctrl.Options{
+	// 		LeaderElection:          c.LeaderElection,
+	// 		LeaderElectionNamespace: c.Namespace,
+	// 		SyncPeriod:              &c.Sync,
+	// 	},
+	// 	ManagerName:      c.Name,
+	// 	ManagerNamespace: c.Namespace,
+	// })
+
+	// approach 2:
+	// keep calling `ctrl.NewManager` with additional config here
+	// ----------------------------------------------------------
+	if c.Name == "" {
+		return errors.New(errManagerName)
+	}
+	if c.Namespace == "" {
+		return errors.New(errManagerNamespace)
+	}
```

Comment:
I believe if we omit the `LeaderElectionNamespace` per my comment below we'll be able to avoid these empty string checks; I'm pretty sure `Name` will never be empty unless there's a programming error.